src/blockdev.rs

```diff
@@ -250,6 +276,51 @@ pub fn udev_settle() -> Result<()> {
     Ok(())
 }
 
+/// A Reader of a disk image that fails if the formatted sector size of the
+/// image doesn't match a specified value.
+pub struct DiskSectorSizeReader<R: Read> {
+    source: BufReader<R>,
+}
+
+impl<R: Read> DiskSectorSizeReader<R> {
+    pub fn new(source: R, expected_size: u32) -> Result<Self> {
```

Comment:
If it doesn't bother you too much, I'd add a quick regression-test covering the 512/4096/garbage cases.